tests/unit/test_redfish.py

```diff
@@ -555,11 +565,69 @@ def mock_get_response(uri):
         mock_redfish_obj.get.assert_any_call("/redfish/v1/Systems/s1/Storage/STOR1")
         mock_redfish_obj.get.assert_any_call("/redfish/v1/Systems/s1/Storage/STOR2")
 
+    @patch("prometheus_hardware_exporter.collectors.redfish.redfish_client")
+    @patch(
+        "prometheus_hardware_exporter.collectors.redfish.redfish_utilities.collections.get_collection_ids"  # noqa: E501
+    )
+    @patch(
+        "prometheus_hardware_exporter.collectors.redfish.redfish_utilities.systems.get_system_ids"
+    )
+    def test_non_standard_storage_uri_name(
+        self, mock_get_system_ids, mock_get_collection_ids, mock_redfish_client
+    ):
+        """Test non-standard name for "Storage" in URI for storage controller and drives.
+
+        Eg: /redfish/v1/Systems/S1/Storages
+        """
+        mock_redfish_obj = Mock()
+        mock_get_system_ids.return_value = ["s1"]
+        mock_get_collection_ids.return_value = ["STOR1"]
+        mock_redfish_client.return_value = mock_redfish_obj
+
+        def mock_get_response(uri):
+            response = Mock()
+            if "Systems/s1/Storages/STOR1/Drives/d11" in uri:
+                response.dict = {
+                    "Id": "d11",
+                    "Status": {"Health": "OK", "State": "Enabled"},
+                }
+            elif "Systems/s1/Storages/STOR1" in uri:
+                response.dict = {
+                    "StorageControllers": [
+                        {
+                            "MemberId": "sc0",
+                            "Status": {"Health": "OK", "State": "Enabled"},
+                        }
+                    ],
+                    "Drives": [
+                        {"@odata.id": "/redfish/v1/Systems/s1/Storages/STOR1/Drives/d11"},
+                    ],
+                }
+            # response for GET request to /redfish/v1/Systems/<sys_id>/
+            elif "Systems" in uri:
+                response.dict = {"Storage": {"@odata.id": "/redfish/v1/Systems/sX/Storages"}}
+            return response
+
+        mock_redfish_obj.get.side_effect = mock_get_response
+
+        # storage controller
+        with RedfishHelper(Mock()) as helper:
+            _sc_count, _sc_data = helper.get_storage_controller_data()
+        mock_get_system_ids.assert_called_once_with(mock_redfish_obj)
+        # test if "Storages" is present in redfish GET request URI
+        mock_redfish_obj.get.assert_any_call("/redfish/v1/Systems/s1/Storages/STOR1")
+
+        # storage drives
+        with RedfishHelper(Mock()) as helper:
+            _drive_count, _drive_data = helper.get_storage_drive_data()
+        # test if "Storages" is present in redfish GET request URI
+        mock_redfish_obj.get.assert_any_call("/redfish/v1/Systems/s1/Storages/STOR1/Drives/d11")
```

Comment:
I prefer to check the expected output rather than checking the url.

Reply:
Changed in the latest patch.